build/node-feature-discovery/Dockerfile

```diff
@@ -0,0 +1,33 @@
+FROM golang:1.13.5-alpine as build
+
+ARG VERSION=v0.6.0
+ARG TARGETPLATFORM
+ENV GO111MODULE=on \
+    CGO_ENABLED=0
+
+RUN apk add make git
+
+WORKDIR /go/node-feature-discovery
+
+RUN git clone --depth 1 -b ${VERSION} https://github.com/kubernetes-sigs/node-feature-discovery.git . \
+    && export GOOS=$(echo ${TARGETPLATFORM} | cut -d / -f1) \
+    && export GOARCH=$(echo ${TARGETPLATFORM} | cut -d / -f2) \
+    && GOARM=$(echo ${TARGETPLATFORM} | cut -d / -f3); export GOARM=${GOARM:1}
+
+RUN go mod download
+
+RUN go install \
+  -ldflags "-s -w -X sigs.k8s.io/node-feature-discovery/pkg/version.version=$VERSION -X sigs.k8s.io/node-feature-discovery/source.pathPrefix=$HOSTMOUNT_PREFIX" \
+  ./cmd/*
+RUN install -D -m644 nfd-worker.conf.example /etc/kubernetes/node-feature-discovery/nfd-worker.conf
+
+RUN make test
+
+# Create production image for running node feature discovery
+FROM debian:stretch-slim
+
+# Use more verbose logging of gRPC
+ENV GRPC_GO_LOG_SEVERITY_LEVEL="INFO"
+
+COPY --from=build /etc/kubernetes/node-feature-discovery /etc/kubernetes/node-feature-discovery
+COPY --from=build /go/bin/nfd-* /usr/bin/
```

Comment:
This should also work for you, see how you go though and what you think, I wouldn't worry about running `make test` because we can assume they test on their release already - also I removed the `go mod download` benefits because they're only when you're building a lot but in our case we only build once per release so just less to worry about.

```suggestion
FROM golang:1.13.5 as builder

ARG VERSION=v0.6.0
ARG TARGETPLATFORM
ARG HOSTMOUNT_PREFIX

ENV GO111MODULE=on \
  CGO_ENABLED=0

WORKDIR /go/node-feature-discovery

RUN apk add git

RUN export GOOS=$(echo ${TARGETPLATFORM} | cut -d / -f1) && \
  export GOARCH=$(echo ${TARGETPLATFORM} | cut -d / -f2) && \
  GOARM=$(echo ${TARGETPLATFORM} | cut -d / -f3 | cut -c2-) && \
  git clone --depth 1 -b ${VERSION} https://github.com/kubernetes-sigs/node-feature-discovery.git . && \
  go install \
  -ldflags "-s -w -X sigs.k8s.io/node-feature-discovery/pkg/version.version=$VERSION -X sigs.k8s.io/node-feature-discovery/source.pathPrefix=$HOSTMOUNT_PREFIX" \
  ./cmd/*

RUN install -D -m644 nfd-worker.conf.example /etc/kubernetes/node-feature-discovery/nfd-worker.conf

FROM debian:stretch-slim

ENV GRPC_GO_LOG_SEVERITY_LEVEL="INFO"

COPY --from=builder /etc/kubernetes/node-feature-discovery /etc/kubernetes/node-feature-discovery
COPY --from=builder /go/bin/nfd-* /usr/bin/
```

Reply:
Fantastic, thank you!  Should the `FROM golang:1.13.5 as builder` be `FROM golang:1.13.5-alpine as build` as we're doing `apk add...` commands?

Reply:
Good spot! I think you're right there!